subcommands/keys/ca_create.go

```diff
@@ -0,0 +1,137 @@
+		Long: `Peform a one-time operation to set up PKI infrastructure for managing
+the device gateway. This command creates a few things:
+
+# Root of trust for your factory: factory_ca.key / factory_ca.pem
+The factory_ca keypair is generated by this command to define the PKI root of
+trust for this factory.
+ * factory_ca.key - An EC prime256v1 private key that should be STORED OFFLINE.
+ * factory_ca.pem - The public x509 certificate that is shared with
+                    Foundries.io. Once set, all future PKI related changes will
+                    require proof you own this certificate.
+
+# online-ca - A Foundries.io owned keypair to support lmp-device-register
+In order for lmp-device-register to work, Foundries.io needs the ability to
+create client certificates for devices. If enabled, the factory_ca keypair will
+sign the certificate signing request returned from the API.
+
+This is optional.
+
+# local-ca - A keypair you own
+This keypair can be used for things like your manufacturing process where you
+may set up devices without having to communicate with Foundries.io web
+services. This keypair is capable of create client certificates for devices.
+If enabled, the local-ca.pem will be shared with the Foundries.io device gateway
+so that it will trust the client certificate of devices created with this
+keypair.
+
+This is optional.`,
+	}
+	caCmd.AddCommand(cmd)
+	cmd.Flags().BoolVarP(&createOnlineCA, "online-ca", "", true, "Create an online CA owned by Foundries that works with lmp-device-register")
+	cmd.Flags().BoolVarP(&createLocalCA, "local-ca", "", true, "Create a local CA that you can use for creating your own device certificates")
+	cmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Create key on an HSM using this module.")
+	cmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to set up on the HSM, if using one.")
+	cmd.Flags().StringVarP(&hsmSoPin, "hsm-so-pin", "", "", "The PKCS#11 Security Officer PIN to set up on the HSM, if using one.")
+}
+
+func writeFile(filename, contents string, mode os.FileMode) {
+	if err := ioutil.WriteFile(filename, []byte(contents), mode); err != nil {
+		fmt.Printf("ERROR: Creating %s: %s", filename, err)
+		os.Exit(1)
+	}
+}
+
+func run(name string, arg ...string) string {
+	cmd := exec.Command(name, arg...)
+	cmd.Stderr = os.Stderr
+	out, err := cmd.Output()
+	subcommands.DieNotNil(err)
+	return string(out)
+}
+
+func doCreateCA(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	certsDir := args[0]
+	logrus.Debugf("Create CA for %s under %s", factory, certsDir)
+	subcommands.DieNotNil(os.Chdir(certsDir))
+
+	if len(hsmModule) > 0 {
+		fmt.Println("Creating pkcs11 token")
+		run("pkcs11-tool", "--module", hsmModule, "--init-token", "--label=device-gateway-root", "--so-pin", hsmSoPin)
+		run("pkcs11-tool", "--module", hsmModule, "--init-pin", "--token-label=device-gateway-root", "--so-pin", hsmSoPin, "--pin", hsmPin)
```

Comment:
For example, we can assume the token and pin is already initialized, and only set the env for the module, token id and pin number, which is all that is required for us to manipulate the token.